### PR TITLE
🐛Fix script tag validation issues with amp-experiment usage on amp-story

### LINKED ISSUE #24748: In amp-story, amp-experiment is alerted by amp validator, but it work correctly.
## What's the issue?
In amp stories, amp-experiment component should be allowed to use,  but it is alerted by amp validator.

Related Link
- https://amp.dev/documentation/components/amp-experiment/?format=stories
- https://amp.dev/documentation/components/amp-story/#children

| contained in amp-story | contained in amp-story-grid-layer | 
| :--- | :--- |
| Tag 'amp-experiment' is disallowed as child of tag 'amp-story'. Child tag must be one of ['amp-analytics', 'amp-consent', 'amp-geo', 'amp-pixel', 'amp-sidebar', 'amp-story-access', 'amp-story-auto-ads', 'amp-story-bookend', 'amp-story-page']. | The tag 'script' may not appear as a descendant of tag 'amp-story-grid-layer'. | 

However, amp-experiment work correctly.
amp-x-ui-experiment attributes are given to body

## How do we reproduce the issue?

1. add amp-experiment component url
2. Following [this example](https://amp.dev/documentation/components/amp-experiment/?format=stories), write amp-experiment  contained in amp-story or amp-story-grid-layer.

## What browsers are affected?

[chrome extension](https://chrome.google.com/webstore/detail/amp-validator/nmoffdblmcmgeicmolmhobpoocbbmknc?hl=ja) version 1.1.4
[amphtml-validator](https://www.npmjs.com/package/amphtml-validator)

## Which AMP version is affected?
1909181902540

### PR DESCRIPTION
**Pull request by jackbsteinberg**
Fixes a bug where the amp-validator would flag script tags used inside amp-experiment tags would flag the validator for `[CUSTOM_JAVASCRIPT_DISALLOWED]`.
Resolves #24748

### CODE REVIEW

File: extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                          }
+                        ]
+                      }
+                    }
+                  }
+                }
+            </script>
+        </amp-experiment>
+
+        <!-- bare script invalid under amp-story-grid-layer -->
```
Comment (honeybadgerdontcare) [inline extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html]: We tend to label comments like these with the expected validation result first then the reason. E.g. `<-- Invalid: bare script not allowed as a child of amp-story-grid-layer -->`. Please update other comments in the file.

Comment (jackbsteinberg) [reply extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html]: Thanks for the heads up, fixed!

**Comment by honeybadgerdontcare** [comment]
@banaag for review

File: extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html (hunk 1 above)
Comment (jackbsteinberg) [reply extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html]: Fixed

File: extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                "yellow": 49,
+                "red": 49
+                }
+              }
+            }
+        </script>
+    </amp-experiment>
+  </amp-story>
+</body>
+</html>
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.html]: Nit: newline here and in the `.out`

File: extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.out [hunk 3]
```diff
@@ -0,0 +1,82 @@
+FAIL
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/validator-amp-story-amp-experiment.out]: We name the `FAIL` test with the suffix `*-error.html` and the `PASS` test without the suffix. 

Could you add a passing test and rename these two files with the `-error` suffix please?

**Comment by gmajoulet** [comment]
Friendly ping @banaag

**Pull request merged**